Replicator/Replicator.cc

```diff
@@ -310,6 +310,14 @@ namespace litecore { namespace repl {
 
 
     void Replicator::onError(C4Error error) {
+        if(status().error.code != 0 && error.domain == WebSocketDomain &&
+           (error.code == kWebSocketCloseAppPermanent || error.code == kWebSocketCloseAppTransient)) {
+            // CBL-1178: If we already have an error code, it is more relevant than the web socket close code, so keep it
+            // intact so that the consumer can know what went wrong
+            logVerbose("kWebSocketCloseAppPermanent or kWebSocketCloseAppTransient received, ignoring (only relevant for underlying connection...)");
+            return;
+        }
+        
```

Comment:
i thought Listener will throw a 403, which will still send same `kCloseAppPermanent` error right? Since this handling is done for only `kWebSocketCloseAppPermanent` and `kWebSocketCloseAppTransient`? 

https://github.com/couchbase/couchbase-lite-core/blob/b0d1ffe8808bb71246d8736f1b5c7d5ecf7e82d0/Replicator/Replicator.cc#L726

Reply:
i got it, thanks